base_report_to_printer/report.py

```diff
@@ -53,6 +71,9 @@ def get_pdf(self, cr, uid, ids, report_name, html=None,
         report = self._get_report_from_name(cr, uid, report_name)
         behaviour = report.behaviour()[report.id]
         printer = behaviour['printer']
-        if behaviour['action'] == 'server' and printer and document:
+        can_print_report = self._can_print_report(cr, uid, ids,
+                                                  behaviour, printer, document,
+                                                  context=context)
+        if can_print_report:
```

Comment:
The alteration of the context was pointless: the context was never used afterwards.
If your idea was to change it so it goes up to the caller, I think it's anyway a bad idea and that's the caller's work to ensure it calls the method correctly. By the way such alteration up to the caller won't be possible with the new API where the context is a frozendict.